drafts/inlinemacros.md

```diff
@@ -80,39 +86,37 @@ by-value parameters and that are not side-effect free paths. Arguments to by-nam
 parameters are left alone.
 
    These transformation are intended to preserve the semantics
-of function applications under inlining: An  function call should have the same
-semantics independently on whether the function was maked `inline` or not.
+of function applications under inlining: A function call should have the same
+semantics with respect to side effects independently on whether the function was made `inline` or not.
+
+   After the rewriting is complete, the result of the rewriting is forcibly upcast
+to the type of the original expression. This mechanism works exactly the same 
+as the mechanism of blackbox macro expansion in Scala 2.10+.
```

Comment:
I think we should outright prohibit any type of whitebox expansion before we implement blackbox expansion.